CHANGELOG.md

```diff
@@ -4,6 +4,14 @@ This file contains notable changes (e.g. breaking changes, major changes, etc.)
 
 This file was introduced starting Kani 0.23.0, so it only contains changes from version 0.23.0 onwards.
 
+## [0.43.0]
+
+## What's Changed
+* Update the rust toolchain to nightly-2023-12-03 by @tautschnig in https://github.com/model-checking/kani/pull/2913
```

Comment:
Sorry, we merge all of the toolchain upgrades into one line and list everyone that contributed to them. Please see previous releases for an example.

Also, in that line we list the current toolchain version.

Reply:
Updated as suggested.